src/components/addOrUpdatePublicationForm/AddOrUpdatePublicationForm.tsx

```diff
@@ -0,0 +1,198 @@
+import React from 'react';
+import { Formik } from 'formik';
+import DialogContent from '@mui/material/DialogContent';
+import {
+  Box,
+  Chip,
+  CircularProgress,
+  Stack,
+  TextField,
+  Typography,
+} from '@mui/material';
+import { grey } from '@mui/material/colors';
+import DialogActions from '@mui/material/DialogActions';
+import Button from '@mui/material/Button';
+import { PublicationModel } from '../../models/Publication.ts';
+import { useCreatePublicationMutation, useUpdatePublicationMutation } from '../../api/PublicationApi.ts';
+
+interface AddOrUpdatePublicationProps {
+  openedPublication?: PublicationModel;
+  housingAssociationId: string;
+  closeDialogFunction: () => void;
+}
+
+interface AddPublicationFormProps {
+  title: string;
+  content: string;
+}
+
+const AddOrUpdatePublicationForm: React.FC<AddOrUpdatePublicationProps> = ({
+                                                                             openedPublication,
+                                                                             housingAssociationId,
+                                                                             closeDialogFunction,
+                                                                           }) => {
+  const validateFields = (values: AddPublicationFormProps) => {
```

Comment:
create a special file for creating validators so we can reuse it later in different parts of code